src/tests/nativeaot/SmokeTests/PInvoke/PInvokeNative.cpp

```diff
@@ -265,6 +283,63 @@ DLL_EXPORT int __stdcall VerifyUnicodeStringRef(unsigned short **val)
     return 1;
 }
 
+DLL_EXPORT int __stdcall VerifyBSTRString(unsigned short *val)
+{
+    if (val == NULL)
+        return 0;
+
+    if (*((int*)val - 1) != 22)
+        return 0;
+
+    unsigned short expected[] = { 'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd', 0};
+
+    return CompareUnicodeString(val, expected);
+}
+
+DLL_EXPORT int __stdcall VerifyBSTRStringOut(unsigned short **val)
+{
+    if (val == NULL)
+        return 0;
+    unsigned short *p = (unsigned short *)MemAlloc(sizeof(unsigned short) * 14);
```

Comment:
It would be more correct to use `SysAllocString` / `SysFreeString` on Windows, and have stub implementations of `SysAllocString` / `SysFreeString` methods for Unix.

Alternatively, I think it would be ok to omit the BSTR tests from the NativeAOT smoke test. We have comprehensive BSTR tests in the regular test suite, and the NativeAOT-specific implementation is simple enough and unlikely to break.